var/spack/repos/builtin/packages/py-gitpython/package.py

```diff
@@ -10,19 +10,23 @@ class PyGitpython(PythonPackage):
     """GitPython is a python library used to interact with Git repositories."""
 
     homepage = "http://gitpython.readthedocs.org"
-    pypi = "gitpython/GitPython-3.1.7.tar.gz"
+    pypi = "GitPython/GitPython-3.1.12.tar.gz"
 
-    version('3.1.8', sha256='49d4f859bda7bd6ef9ce72d9a2cddb6a59dd70f34a18ce409670bc655431c517')
-    version('3.1.6', sha256='00d459b1fd2d4a2106252867701f7303485654bdb115275fa6f788756d18879e')
-    version('3.1.5', sha256='2f7f9716afecc099bebf4d1a5e4a7df96f32c8255254e83642d8acf70f1b82c3')
-    version('3.1.4', sha256='946e9c1daa755b8d4aaaad7f96c3fb4ac99ba36379b4355d6a3a4966ba68ca12')
-    version('3.1.3', sha256='12da19250d1db60a9562780c3ecd8a9713da59ec69a3e207db2af1a0afe87c96')
-    version('3.1.2', sha256='ce862c2c5e569728eae2d2441d358ab99a2492c7418cb3bdba1517c46a625cc1')
-    version('3.1.1', sha256='76f35b8d99f02432632d33c0d64419c65d04377cb68b80bfc1819c421a3c05ca')
-    version('3.1.0', sha256='a7d97b664ac61474147db23671aa8cc12be16248a405e3db15538d619d15cb51')
-    version('3.0.9', sha256='d92bed79c12ce4198efe34337c55d85984def3d648ea465049e0ec6935e23c21')
-    version('0.3.6', sha256='e6599fcb939cb9b25a015a429702db39de10f2b493655ed5669c49c37707d233')
+    version('3.1.12',  sha256='42dbefd8d9e2576c496ed0059f3103dcef7125b9ce16f9d5f9c834aed44a1dac')
+    version('3.1.11',  sha256='befa4d101f91bad1b632df4308ec64555db684c360bd7d2130b4807d49ce86b8')
+    version('3.1.10',  sha256='f488d43600d7299567b59fe41497d313e7c1253a9f2a8ebd2df8af2a1151c71d')
+    version('3.1.9',  sha256='a03f728b49ce9597a6655793207c6ab0da55519368ff5961e4a74ae475b9fa8e')
+    version('3.1.8',  sha256='080bf8e2cf1a2b907634761c2eaefbe83b69930c94c66ad11b65a8252959f912')
+    version('3.1.7',  sha256='2db287d71a284e22e5c2846042d0602465c7434d910406990d5b74df4afb0858')
+    version('3.1.6',  sha256='b54969b3262d4647f80ace8e9dd4e3f99ac30cc0f3e766415b349208f810908f')
+    version('3.1.5',  sha256='90400ecfa87bac36ac75dfa7b62e83a02017b51759f6eef4494e4de775b2b4be')
+    version('3.1.4',  sha256='fa98ce1f05805d59bbc3adb16c0780e5ca43b5ea9422feecf1cd0949a61d947e')
+    version('3.1.3',  sha256='e107af4d873daed64648b4f4beb89f89f0cfbe3ef558fc7821ed2331c2f8da1a')
+    version('3.1.2',  sha256='864a47472548f3ba716ca202e034c1900f197c0fb3a08f641c20c3cafd15ed94')
+    version('3.1.1',  sha256='6d4f10e2aaad1864bb0f17ec06a2c2831534140e5883c350d58b4e85189dab74')
+    version('3.1.0',  sha256='e426c3b587bd58c482f0b7fe6145ff4ac7ae6c82673fc656f489719abca6f4cb')
+    version('3.0.9',  sha256='7e5df21bfef38505115ad92544fb379e6fedb2753f3b709174c4358cecd0cb97')
 
-    depends_on('python@3.4:',   type=('build', 'run'))
+    depends_on('python@3.5:',   type=('build', 'run'))
```

Comment:
3.12 still supports Python 3.4, I would undo this change.

Reply:
I was running into an error with setuptools:

```
spack install py-gitpython ^python@3.4
==> Warning: Failed to initialize repository: '/home/joshuasbrown/Software/kokkos-spack/kokkos'.
  No repo.yaml found in '/home/joshuasbrown/Software/kokkos-spack/kokkos'
  To remove the bad repository, run this command:
      spack repo rm /home/joshuasbrown/Software/kokkos-spack/kokkos
==> Error: An unsatisfiable version constraint has been detected for spec:

    python@3.4%gcc@7.4.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib arch=linux-centos7-haswell
        ^bzip2
            ^diffutils
                ^iconv
        ^expat
        ^gdbm
            ^readline
                ^ncurses
                    ^pkgconf@1.7.3%gcc@7.4.0 arch=linux-centos7-haswell
        ^gettext+libxml2
            ^libxml2
                ^xz
                ^zlib@1.1.3:
        ^libffi
        ^openssl@:1.0.2z
            ^perl@5.14.0:
                ^berkeley-db
        ^sqlite@3.0.8:
        ^uuid


while trying to concretize the partial spec:

    py-setuptools@50.3.2%gcc@7.4.0 arch=linux-centos7-haswell


py-setuptools requires python version 3.5:, but spec asked for 3.4
```

Reply:
Maybe I need to use an earlier version of setup tools for python3.4, I'm not quite sure how to specify that. I also would have thought that dependency should be resolved in the setuptools python package.py.

Reply:
You have a couple of options. Your best bet is to use the latest/default version of Python and not use 3.4 cuz it's ancient. If you really want to use 3.4, you can do:
```console
$ spack spec py-gitpython ^python@:3.4 ^py-setuptools@:43
```

Reply:
ok.